[BUG] - SSL Certificate Verification Failure in Dask Gateway Unit Tests

### Describe the bug

Our unit tests have begun failing due to SSL certificate verification errors when connecting to the Dask Gateway. Specifically, we encounter an SSLCertVerificationError caused by a self-signed certificate. 
```shell
=================================== FAILURES ===================================
______________________________ test_dask_gateway _______________________________
....
  File "/home/runnerx/miniconda3/envs/test/lib/python3.11/site-packages/dask_gateway/client.py", line 397, in _request
    resp = await session.request(method, url, json=json, **self._request_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runnerx/miniconda3/envs/test/lib/python3.11/site-packages/aiohttp/client.py", line [659](https://github.com/nebari-dev/nebari/actions/runs/11022572877/job/30612182623?pr=2739#step:16:660), in _request
    conn = await self._connector.connect(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runnerx/miniconda3/envs/test/lib/python3.11/site-packages/aiohttp/connector.py", line 557, in connect
    proto = await self._create_connection(req, traces, timeout)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runnerx/miniconda3/envs/test/lib/python3.11/site-packages/aiohttp/connector.py", line 1002, in _create_connection
    _, proto = await self._create_direct_connection(req, traces, timeout)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runnerx/miniconda3/envs/test/lib/python3.11/site-packages/aiohttp/connector.py", line 1336, in _create_direct_connection
    raise last_exc
  File "/home/runnerx/miniconda3/envs/test/lib/python3.11/site-packages/aiohttp/connector.py", line 1305, in _create_direct_connection
    transp, proto = await self._wrap_create_connection(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runnerx/miniconda3/envs/test/lib/python3.11/site-packages/aiohttp/connector.py", line 1070, in _wrap_create_connection
    raise ClientConnectorCertificateError(req.connection_key, exc) from exc
aiohttp.client_exceptions.ClientConnectorCertificateError: Cannot connect to host github-actions.nebari.dev:443 ssl:True [SSLCertVerificationError: (1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate (_ssl.c:1006)')]
```
We've been using a monkey patch to disable SSL verification as a temporary workaround, which has allowed tests to pass so far; however, this does not work anymore. 
https://github.com/nebari-dev/nebari/blob/b2afe1a21c7ba03940f8df360ddced24d5ee08f6/tests/tests_deployment/utils.py#L115-L132

My guess, based on a similar encounter, is that aiohttp the base lib used by dask to handle the connections has changed something internally regarding the [TCPConnector](https://docs.aiohttp.org/en/stable/client_reference.html#aiohttp.TCPConnector) default behavior (this happens internally within the requests, and is decided by a combination of parameters that dask handles based on the provided proxy schema, which in our case we were passing for jupyterhub authentication)

```python
    async def _request(self, method, url, json=None):
        if self._session is None:
            # "unsafe" allows cookies to be set for ip addresses, which can
            # commonly serve dask-gateway deployments. Since this client is
            # only ever used with a single endpoint, there is no danger of
            # leaking cookies to a different server that happens to have the
            # same ip.
            self._session = aiohttp.ClientSession(
                cookie_jar=aiohttp.CookieJar(unsafe=True)
            )
        session = self._session

        resp = await session.request(method, url, json=json, **self._request_kwargs)
```
To address this adequately, I propose overriding the `dask_gateway.Gateway` class  using `self._request_kwargs` object above to allow the TLS parameter to be disabled within our test context. 

### Expected behavior

- unit tests should run correctly

### OS and architecture in which you are running Nebari

Linux (CI/GHA)

### How to Reproduce the problem?

https://github.com/nebari-dev/nebari/actions/runs/11022572877/job/30612182623?pr=2739

### Command output

_No response_

### Versions and dependencies used.

_No response_

### Compute environment

kind

### Integrations

_No response_

### Anything else?

_No response_

--- a/tests/tests_deployment/test_dask_gateway.py
+++ b/tests/tests_deployment/test_dask_gateway.py
@@ -4,7 +4,7 @@ import dask_gateway
 import pytest
 
 from tests.tests_deployment import constants
-from tests.tests_deployment.utils import get_jupyterhub_token, monkeypatch_ssl_context
+from tests.tests_deployment.utils import get_jupyterhub_token
 
 # monkeypatch_ssl_context()
 

--- a/tests/tests_deployment/test_dask_gateway.py
+++ b/tests/tests_deployment/test_dask_gateway.py
@@ -6,7 +6,7 @@ import pytest
 from tests.tests_deployment import constants
 from tests.tests_deployment.utils import get_jupyterhub_token, monkeypatch_ssl_context
 
-monkeypatch_ssl_context()
+# monkeypatch_ssl_context()
 
 
 @pytest.fixture
@@ -15,7 +15,17 @@ def dask_gateway_object():
     os.environ["JUPYTERHUB_API_TOKEN"] = get_jupyterhub_token(
         "dask-gateway-pytest-token"
     )
-    return dask_gateway.Gateway(
+
+    # Create custom class from Gateway that disables the tls/ssl verification
+    # to do that we will override the self._request_kwargs dictionary within the
+    # __init__, targeting aiohttp.ClientSession.request method
+
+    class DaskGateway(dask_gateway.Gateway):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+            self._request_kwargs.update({"ssl": False})
+
+    return DaskGateway(
         address=f"https://{constants.NEBARI_HOSTNAME}/{constants.GATEWAY_ENDPOINT}",
         auth="jupyterhub",
         proxy_address=f"https://{constants.NEBARI_HOSTNAME}:8786",

--- a/tests/tests_deployment/test_dask_gateway.py
+++ b/tests/tests_deployment/test_dask_gateway.py
@@ -6,8 +6,6 @@ import pytest
 from tests.tests_deployment import constants
 from tests.tests_deployment.utils import get_jupyterhub_token
 
-# monkeypatch_ssl_context()
-
 
 @pytest.fixture
 def dask_gateway_object():

--- a/tests/tests_deployment/test_dask_gateway.py
+++ b/tests/tests_deployment/test_dask_gateway.py
@@ -18,7 +18,7 @@ def dask_gateway_object():
     return dask_gateway.Gateway(
         address=f"https://{constants.NEBARI_HOSTNAME}/{constants.GATEWAY_ENDPOINT}",
         auth="jupyterhub",
-        proxy_address=f"tcp://{constants.NEBARI_HOSTNAME}:8786",
+        proxy_address=f"https://{constants.NEBARI_HOSTNAME}:8786",
     )
 
 

--- a/tests/tests_deployment/test_dask_gateway.py
+++ b/tests/tests_deployment/test_dask_gateway.py
@@ -26,7 +26,7 @@ def dask_gateway_object():
     return DaskGateway(
         address=f"https://{constants.NEBARI_HOSTNAME}/{constants.GATEWAY_ENDPOINT}",
         auth="jupyterhub",
-        proxy_address=f"https://{constants.NEBARI_HOSTNAME}:8786",
+        proxy_address=f"tcp://{constants.NEBARI_HOSTNAME}:8786",
     )
 
 

